operations_on_datastructures/linked_list_right_rotation.cpp

```diff
@@ -0,0 +1,215 @@
+    int linked_list::countElements()
+    {
+        int count = 0;
+        std::shared_ptr<Node> temp = head; ///< pointer to iterate through linked list
+        while (temp->next != nullptr)
+        {
+            count++;
+            temp = temp->next;
+        }
+        return count;
+    }
+
+    /**
+     * function adds new element to the end of the linked list
+     * @param d to be added to the end of the linked list
+     */
+    void linked_list::push_back(int d)
+    {
+        if (isEmpty())
+        {
+            head->next = std::make_shared<Node>(d);
+            tail = head->next;
+        }
+        else
+        {
+            tail->next = std::make_shared<Node>(d);
+            tail = tail->next;
+        }
+    }
+
+    /**
+     * function rotates linked list to the right by given n times
+     * @param n times to be rotated
+     * @returns 'void'
+     */
+    void linked_list::rotate(int n)
+    {
+
+        int length = countElements(); ///< variable for total number of elements
+
+        /**
+          *  if the linked list is rotated as many times as its total number of elements,
+             then there is no change in linked list.
+        */
+        if (n >= length)
+        {
+            /**
+             * eliminate the number of the times the linked list will be rotated back to
+               the original linked list
+             */
+            n = n % length;
+        }
+
+        if (n == 0){
+            display();
+            return; ///< if n beacomes zero; after rotating n times linked list wont be changed
+        }
+
+        std::shared_ptr<Node> curr = head; ///< pointer to iterate through linked list
+        std::shared_ptr<Node> newtail;     ///< pointer for the new tail to be made
+
+        /// loop for iterationg pointer to the nth element (to be the new head)
+        for (int i = 0; i < length-n; ++i)
+        {
+            curr = curr->next;
+        }
+        newtail = curr;
+
+        /// loop for iterating pointer to the end of linked list
+        while (curr->next != NULL)
+        {
+            curr = curr->next;
+        }
+        curr->next = head->next;
+        curr = newtail->next;
+        newtail->next = NULL;
+        head->next = curr;
+    }
+
+    /**
+     * function displays all the elements in the linked list
+     * @returns 'void'
+     */
+    void linked_list::display()
+    {
+        if (isEmpty())
+        {
+            std::cout << "List is Empty!";
+            return;
+        }
+        std::shared_ptr<Node> t = head; ///< pointer to iterate through linked list
+        while (t->next != nullptr)
+        {
+            std::cout << t->next->data << " ";
+            t = t->next;
+        }
+        std::cout << "\n";
+    }
+
+}  // namespace operation_on_datastructures
+
+/**
+ * @brief Main function
+ * @param argc commandline argument count (ignored)
+ * @param argv commandline array of arguments (ignored)
+ * @returns 0 on exit
+ */
+
```

Comment:
Please add self-test cases so we can verify your program works properly (**see the [typical structure of a program](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#typical-structure-of-a-program) as a reference**). Let us know if you need any help here or on our [**Discord server**](https://the-algorithms.com/discord). 😃